diff --git a/base/BUILD.gn b/base/BUILD.gn
index 092acd0d06e5a0d1a4e44c4809ec4e8a838d7ee2..fd005e5f7040cdec930838d5f5a932ef36a014d4 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -3897,6 +3897,7 @@ if (is_android) {
       "android/java/src/org/chromium/base/task/TaskTraits.java",
       "android/java/src/org/chromium/base/task/TaskTraitsExtensionDescriptor.java",
     ]
+    sources += brave_java_base_sources
 
     annotation_processor_deps = [ "//base/android/jni_generator:jni_processor" ]
 
