diff --git a/build/config/BUILDCONFIG.gn b/build/config/BUILDCONFIG.gn
index 4bedf1ef1b3c642ff2839d0ad62caaedd1e1254f..e898e0518f42794a6417113d92da6aa12353705d 100644
--- a/build/config/BUILDCONFIG.gn
+++ b/build/config/BUILDCONFIG.gn
@@ -314,6 +314,7 @@ is_posix = !is_win && !is_fuchsia
 
 # Holds all configs used for running the compiler.
 default_compiler_configs = [
+  "//brave/build:compiler",
   "//build/config:feature_flags",
   "//build/config/compiler:afdo",
   "//build/config/compiler:afdo_optimize_size",
