diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 47853545570fbe7bc5ff34f3716a3f6c1baeecd4..0d52832b869b65948a9e67abc688e6b8a7a9c818 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1691,6 +1691,7 @@ config("chromium_code") {
       cflags += [ "-Wextra-semi" ]
     }
   }
+  cflags -= brave_removed_chromium_code_cflags
 
   configs = [ ":default_warnings" ]
 }
