diff --git a/chrome/tools/build/win/create_installer_archive.py b/chrome/tools/build/win/create_installer_archive.py
index 61759912c424de8b3a1d92a7f9bf4456ca3f02e1..874ce7c27a4babfd90ac0734af3bb520a29705ea 100755
--- a/chrome/tools/build/win/create_installer_archive.py
+++ b/chrome/tools/build/win/create_installer_archive.py
@@ -21,6 +21,7 @@ import shutil
 import subprocess
 import sys
 
+from create_installer_archive_helper import SignAndCopyPreSignedBinaries, BraveCopyAllFilesToStagingDir
 
 ARCHIVE_DIR = "installer_archive"
 
@@ -110,6 +111,7 @@ def CopyAllFilesToStagingDir(config, distribution, staging_dir, build_dir,
                                    staging_dir, build_dir)
   if enable_hidpi == '1':
     CopySectionFilesToStagingDir(config, 'HIDPI', staging_dir, build_dir)
+  BraveCopyAllFilesToStagingDir(config, staging_dir, g_archive_inputs)
 
 
 def CopySectionFilesToStagingDir(config, section, staging_dir, src_dir):
@@ -529,6 +531,7 @@ def main(options):
   if prev_version:
     version_numbers = prev_version.split('.')
     prev_build_number = version_numbers[2] + '.' + version_numbers[3]
+  SignAndCopyPreSignedBinaries(options.skip_signing, options.output_dir, staging_dir, current_version)
 
   # Name of the archive file built (for example - chrome.7z or
   # patch-<old_version>-<new_version>.7z or patch-<new_version>.7z
@@ -598,6 +601,8 @@ def _ParseOptions():
            'with the installer archive {x86|x64}.')
   parser.add_option('-v', '--verbose', action='store_true', dest='verbose',
                     default=False)
+  parser.add_option('--skip_signing', default=False)
+
 
   options, _ = parser.parse_args()
   if not options.build_dir:
