diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java
index a6584c1cdc727226d6aac1e2bc1ccdd87720114e..e7e722ad1ae3b0d6019cf2b11dc1441202fcd202 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/privacy/ClearBrowsingDataFragment.java
@@ -224,7 +224,7 @@ public abstract class ClearBrowsingDataFragment extends PreferenceFragmentCompat
     /**
      * @return All available {@link DialogOption} entries.
      */
-    protected static final Set<Integer> getAllOptions() {
+    protected Set<Integer> getAllOptions() {
         assert DialogOption.CLEAR_HISTORY == 0;
 
         Set<Integer> all = new ArraySet<>();
@@ -235,6 +235,7 @@ public abstract class ClearBrowsingDataFragment extends PreferenceFragmentCompat
     }
 
     static @BrowsingDataType int getDataType(@DialogOption int type) {
+        if (BraveClearBrowsingDataPreferencesAdvanced.isValidDataType(type)) return BraveClearBrowsingDataPreferencesAdvanced.getDataType(type);
         switch (type) {
             case DialogOption.CLEAR_CACHE:
                 return BrowsingDataType.CACHE;
@@ -254,6 +255,7 @@ public abstract class ClearBrowsingDataFragment extends PreferenceFragmentCompat
     }
 
     static String getPreferenceKey(@DialogOption int type) {
+        if (BraveClearBrowsingDataPreferencesAdvanced.isValidDataType(type)) return BraveClearBrowsingDataPreferencesAdvanced.getPreferenceKey(type);
         switch (type) {
             case DialogOption.CLEAR_CACHE:
                 return "clear_cache_checkbox";
@@ -273,6 +275,7 @@ public abstract class ClearBrowsingDataFragment extends PreferenceFragmentCompat
     }
 
     static @DrawableRes int getIcon(@DialogOption int type) {
+        if (BraveClearBrowsingDataPreferencesAdvanced.isValidDataType(type)) return BraveClearBrowsingDataPreferencesAdvanced.getIcon(type);
         switch (type) {
             case DialogOption.CLEAR_CACHE:
                 return R.drawable.ic_collections_grey;
