diff --git a/chrome/android/chrome_java_resources.gni b/chrome/android/chrome_java_resources.gni
index 328c051b392cbbd35a12733a5b75a186323c3578..f01bd377bae753a0d98278b3dcbe5d3bcae66ff0 100644
--- a/chrome/android/chrome_java_resources.gni
+++ b/chrome/android/chrome_java_resources.gni
@@ -94,7 +94,6 @@ chrome_java_resources = [
   "java/res/drawable-hdpi/infobar_translate.png",
   "java/res/drawable-hdpi/jcb_card.png",
   "java/res/drawable-hdpi/location_bar_incognito_badge.png",
-  "java/res/drawable-hdpi/logo_card_back.png",
   "java/res/drawable-hdpi/mc_card.png",
   "java/res/drawable-hdpi/menu_update.png",
   "java/res/drawable-hdpi/modern_location_bar.9.png",
@@ -230,7 +229,6 @@ chrome_java_resources = [
   "java/res/drawable-mdpi/infobar_translate.png",
   "java/res/drawable-mdpi/jcb_card.png",
   "java/res/drawable-mdpi/location_bar_incognito_badge.png",
-  "java/res/drawable-mdpi/logo_card_back.png",
   "java/res/drawable-mdpi/mc_card.png",
   "java/res/drawable-mdpi/menu_update.png",
   "java/res/drawable-mdpi/modern_location_bar.9.png",
@@ -354,7 +352,6 @@ chrome_java_resources = [
   "java/res/drawable-xhdpi/infobar_translate.png",
   "java/res/drawable-xhdpi/jcb_card.png",
   "java/res/drawable-xhdpi/location_bar_incognito_badge.png",
-  "java/res/drawable-xhdpi/logo_card_back.png",
   "java/res/drawable-xhdpi/mc_card.png",
   "java/res/drawable-xhdpi/menu_update.png",
   "java/res/drawable-xhdpi/modern_location_bar.9.png",
