diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
index 833129c7c2647efb26f9c60d1b993a24a36bbc98..695aeaa8584f0545d3b094b4aacc1072f405f946 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -837,6 +837,7 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
     // TODO(crbug.com/1026020): Move this logic to BottomToolbar class.
     private void onShareDelegateAvailable(ShareDelegate shareDelegate) {
         final OnClickListener shareButtonListener = v -> {
+/*
             recordBottomToolbarUseForIPH();
             RecordUserAction.record("MobileBottomToolbarShareButton");
             Tab tab = null;
@@ -847,7 +848,8 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
                 activity = ((TabImpl) tab).getActivity();
                 isIncognito = tab.isIncognito();
             }
-            shareDelegate.share(tab, /*shareDirectly=*/false);
+*///        shareDelegate.share(tab, /*shareDirectly=*/false);
+          mActivity.addOrEditBookmark(mActivity.getActivityTab());
         };
         mShareButtonListenerSupplier.set(shareButtonListener);
     }
@@ -1862,6 +1864,7 @@ public class ToolbarManager implements ScrimObserver, ToolbarTabController, UrlF
         boolean editingAllowed = currentTab == null || mBookmarkBridge == null
                 || mBookmarkBridge.isEditBookmarksEnabled();
         mToolbar.updateBookmarkButton(isBookmarked, editingAllowed);
+        if (mBottomControlsCoordinator != null) {mBottomControlsCoordinator.updateBookmarkButton(isBookmarked, editingAllowed);}
     }
 
     private void updateReloadState(boolean tabCrashed) {
