diff --git a/chrome/browser/resources/extensions/detail_view.js b/chrome/browser/resources/extensions/detail_view.js
index 36602faefa99c089f1c34a1eb5ff178e55524b5c..70fbfb5b1b95240f23dbc8edcffec8364794eb57 100644
--- a/chrome/browser/resources/extensions/detail_view.js
+++ b/chrome/browser/resources/extensions/detail_view.js
@@ -23,6 +23,7 @@ import './shared_style.js';
 import './shared_vars.js';
 import './strings.m.js';
 import './toggle_row.js';
+import './brave_tor_toggle_row.js';
 
 import {CrContainerShadowBehavior} from 'chrome://resources/cr_elements/cr_container_shadow_behavior.m.js';
 import {focusWithoutInk} from 'chrome://resources/js/cr/ui/focus_without_ink.m.js';
