diff --git a/chrome/browser/importer/profile_writer.h b/chrome/browser/importer/profile_writer.h
index a215b61bf8070b9cf9451a414936ebff441a5e95..1fb9d3134dfd6f23332ca938c0cb43860bac6a40 100644
--- a/chrome/browser/importer/profile_writer.h
+++ b/chrome/browser/importer/profile_writer.h
@@ -86,6 +86,7 @@ class ProfileWriter : public base::RefCountedThreadSafe<ProfileWriter> {
   virtual void AddAutofillFormDataEntries(
       const std::vector<autofill::AutofillEntry>& autofill_entries);
 
+  BRAVE_PROFILE_WRITER_H
  protected:
   friend class base::RefCountedThreadSafe<ProfileWriter>;
 
