diff --git a/chrome/android/features/tab_ui/BUILD.gn b/chrome/android/features/tab_ui/BUILD.gn
index 926c8ce934444c69057ff1aa1422c7dd0218b5f0..71bbf68a81ab174ed06d7e8805f132ff15ca316c 100644
--- a/chrome/android/features/tab_ui/BUILD.gn
+++ b/chrome/android/features/tab_ui/BUILD.gn
@@ -189,6 +189,7 @@ android_library("java") {
     "java/src/org/chromium/chrome/browser/tasks/tab_management/suggestions/TabSuggestionsRanker.java",
     "java/src/org/chromium/chrome/browser/tasks/tab_management/suggestions/TabSuggestionsServerFetcher.java",
   ]
+  sources += brave_android_features_tab_ui_java_sources
 
   deps = [
     ":java_resources",
