diff --git a/chrome/browser/resources/tools/rollup_plugin.js b/chrome/browser/resources/tools/rollup_plugin.js
index ca1db0d08aa98e75e18ca5f46e8e23dc09df71d5..fadfa50bd0d1efe9e23e22151712f6efb8718853 100644
--- a/chrome/browser/resources/tools/rollup_plugin.js
+++ b/chrome/browser/resources/tools/rollup_plugin.js
@@ -77,10 +77,27 @@ export default function plugin(rootPath, hostUrl, excludes, externalPaths) {
     name: 'webui-path-resolver-plugin',
 
     resolveId(source, origin) {
+      /*
+      var fs = require('fs');
+      var logStream = fs.createWriteStream('c:\\max\\tmp\\rollup.log', {flags: 'a'});
+      var msg = '\n----------------\n' +
+                'resolveId:' +
+                '\n\tsource: [' +
+                source +
+                ']\n\torigin: [' +
+                origin +
+                '],\n\trootPath: [' +
+                rootPath +
+                '],\n\thostUrl: [' +
+                hostUrl +
+                '],\n';
+      logStream.write(msg);
+      */
       // Normalize origin paths to use forward slashes.
       if (origin) {
         origin = normalizeSlashes(origin);
       }
+      if (source.startsWith('//brave-resources/')) { source = 'chrome://' + source.substring(2); origin = undefined; } else if (source.startsWith('chrome://brave-resources/')) { origin = undefined; }
 
       for (const [url, path] of urlsToPaths) {
         const resultPath = getPathForUrl(source, origin, url, path, excludes);
