diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index de3b455eded016ac567d5716ad9de60ab66f201c..d52d4efe21ce712c4048619a453af95b4badaadd 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -2467,6 +2467,7 @@ static_library("browser") {
       "performance_monitor/metric_evaluator_helper_posix.h",
     ]
   }
+  sources += brave_chrome_browser_sources deps += brave_chrome_browser_deps public_deps += brave_chrome_browser_public_deps
 
   if (is_android) {
     sources += [
