diff --git a/content/browser/frame_host/navigation_request.cc b/content/browser/frame_host/navigation_request.cc
index 207246b03563bc2d242bcdf0e00d7eb8d891a48f..381d08975381c76e72307ea57551049b86515fc1 100644
--- a/content/browser/frame_host/navigation_request.cc
+++ b/content/browser/frame_host/navigation_request.cc
@@ -1085,8 +1085,11 @@ NavigationRequest::NavigationRequest(
   }
 
   // Sanitize the referrer.
+  LOG(ERROR) << __func__;
+  LOG(ERROR) << "Before sanitize: " << common_params_->url << " " << common_params_->referrer->url;
   common_params_->referrer = Referrer::SanitizeForRequest(
       common_params_->url, *common_params_->referrer);
+  LOG(ERROR) << "After sanitize: " << common_params_->url << " " << common_params_->referrer->url;
 
   if (frame_tree_node_->IsMainFrame()) {
     loading_mem_tracker_ =
@@ -1465,11 +1468,13 @@ void NavigationRequest::StartNavigation(bool is_for_commit) {
   if (common_params_->transition & ui::PAGE_TRANSITION_CLIENT_REDIRECT) {
     // If the page contained a client redirect (meta refresh,
     // document.location), set the referrer appropriately.
+    LOG(ERROR) << "Client redirect";
     sanitized_referrer_ = blink::mojom::Referrer::New(
         redirect_chain_[0], Referrer::SanitizeForRequest(
                                 common_params_->url, *common_params_->referrer)
                                 ->policy);
   } else {
+    LOG(ERROR) << "Sanitize in " << __func__;
     sanitized_referrer_ = Referrer::SanitizeForRequest(
         common_params_->url, *common_params_->referrer);
   }
@@ -1811,9 +1816,14 @@ void NavigationRequest::OnRequestRedirected(
   commit_params_->redirects.push_back(common_params_->url);
   common_params_->url = redirect_info.new_url;
   common_params_->method = redirect_info.new_method;
+  LOG(ERROR) << __func__;
+  LOG(ERROR) << "was: " << common_params_->referrer->url << " now: " << GURL(redirect_info.new_referrer);
   common_params_->referrer->url = GURL(redirect_info.new_referrer);
+  common_params_->referrer->url = GURL();
   common_params_->referrer = Referrer::SanitizeForRequest(
       common_params_->url, *common_params_->referrer);
+  LOG(ERROR) << "sanitized: " << common_params_->referrer->url;
+
 
   // On redirects, the initial referrer is no longer correct, so it must
   // be updated.  (A parallel process updates the outgoing referrer in the
@@ -2770,6 +2780,7 @@ void NavigationRequest::OnStartChecksComplete(
   headers.MergeFrom(TakeModifiedRequestHeaders());
   begin_params_->headers = headers.ToString();
 
+  BRAVE_ONSTARTCHECKSCOMPLETE_MAYBEHIDEREFERRER
   // TODO(clamy): Avoid cloning the navigation params and create the
   // ResourceRequest directly here.
   std::vector<std::unique_ptr<NavigationLoaderInterceptor>> interceptor;
