diff --git a/extensions/browser/sandboxed_unpacker.cc b/extensions/browser/sandboxed_unpacker.cc
index 5b7be8bf71ac49f40f666bc2478573b3d9a9112d..f9c8e1c420cbe6cd2a9989d0fad21632fbc6992c 100644
--- a/extensions/browser/sandboxed_unpacker.cc
+++ b/extensions/browser/sandboxed_unpacker.cc
@@ -991,6 +991,7 @@ base::Optional<base::Value> SandboxedUnpacker::RewriteManifestFile(
   DCHECK(!public_key_.empty());
   base::Value final_manifest = manifest.Clone();
   final_manifest.SetStringKey(manifest_keys::kPublicKey, public_key_);
+  BRAVE_SANDBOXEDUNPACKER_REWRITEMANIFESTFILE
 
   {
     std::string differential_fingerprint;
