diff --git a/extensions/browser/extension_registry.h b/extensions/browser/extension_registry.h
index 293b2cd6b57df4e51f9e796015001ccd2f400742..628125602abb0aa369f44332e97fd94fd315aa35 100644
--- a/extensions/browser/extension_registry.h
+++ b/extensions/browser/extension_registry.h
@@ -45,7 +45,8 @@ class ExtensionRegistry : public KeyedService {
     TERMINATED = 1 << 2,
     BLACKLISTED = 1 << 3,
     BLOCKED = 1 << 4,
-    EVERYTHING = (1 << 5) - 1,
+    RECOMMENDED = 1 << 5,
+    EVERYTHING = (1 << 6) - 1,
   };
 
   explicit ExtensionRegistry(content::BrowserContext* browser_context);
@@ -71,6 +72,9 @@ class ExtensionRegistry : public KeyedService {
     return blacklisted_extensions_;
   }
   const ExtensionSet& blocked_extensions() const { return blocked_extensions_; }
+  const ExtensionSet& recommended_extensions() const {
+    return recommended_extensions_;
+  }
   const ExtensionSet& ready_extensions() const { return ready_extensions_; }
 
   // Returns the set of all installed extensions, regardless of state (enabled,
@@ -174,6 +178,10 @@ class ExtensionRegistry : public KeyedService {
   bool AddBlocked(const scoped_refptr<const Extension>& extension);
   bool RemoveBlocked(const std::string& id);
 
+  // As above, but for the recommended set.
+  bool AddRecommended(const scoped_refptr<const Extension>& extension);
+  bool RemoveRecommended(const std::string& id);
+
   // As above, but for the ready set.
   bool AddReady(const scoped_refptr<const Extension>& extension);
   bool RemoveReady(const std::string& id);
@@ -203,6 +211,9 @@ class ExtensionRegistry : public KeyedService {
   // Extensions that are installed and blocked. Will never be loaded.
   ExtensionSet blocked_extensions_;
 
+  // Extensions that are recommended by Brave.
+  ExtensionSet recommended_extensions_;
+
   // Extensions that are ready for execution. This set is a non-exclusive
   // subset of |enabled_extensions_|.
   ExtensionSet ready_extensions_;
