diff --git a/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d.cc b/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d.cc
index 39c7b1340a5e79aac0223116c4618bd3e9acfcb7..3cc8c497b3ef078e8cd27568d569b2daa969c6cb 100644
--- a/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d.cc
+++ b/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d.cc
@@ -34,6 +34,7 @@
 #include "third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d.h"
 
 #include "base/metrics/histogram_functions.h"
+#include "brave/renderer/brave_content_settings_observer_helper.h"
 #include "third_party/blink/public/common/features.h"
 #include "third_party/blink/public/platform/platform.h"
 #include "third_party/blink/public/platform/task_type.h"
@@ -75,6 +76,12 @@
 #include "third_party/blink/renderer/platform/wtf/text/string_builder.h"
 #include "third_party/blink/renderer/platform/wtf/typed_arrays/array_buffer_contents.h"
 
+#include "brave/third_party/blink/brave_page_graph/buildflags/buildflags.h"
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+#include "brave/third_party/blink/brave_page_graph/page_graph.h"
+#include <sstream>
+#endif
+
 namespace blink {
 
 static const base::TimeDelta kTryRestoreContextInterval =
@@ -796,6 +797,9 @@ TextMetrics* CanvasRenderingContext2D::measureText(const String& text) {
   if (!canvas()->GetDocument().GetFrame())
     return MakeGarbageCollected<TextMetrics>();
 
+  if (!AllowFingerprinting(canvas()->GetDocument().GetFrame()))
+    return MakeGarbageCollected<TextMetrics>();
+
   base::TimeTicks start_time = base::TimeTicks::Now();
   canvas()->GetDocument().UpdateStyleAndLayoutTreeForNode(canvas());
 
@@ -808,6 +812,44 @@ TextMetrics* CanvasRenderingContext2D::measureText(const String& text) {
   else
     direction = ToTextDirection(GetState().GetDirection(), canvas());
 
+#if BUILDFLAG(BRAVE_PAGE_GRAPH_ENABLED)
+  {
+    TextMetrics* result = MakeGarbageCollected<TextMetrics>(font, direction,
+        GetState().GetTextBaseline(), GetState().GetTextAlign(), text);
+
+    brave_page_graph::PageGraph* page_graph =
+      canvas()->GetDocument().GetPageGraph();
+    if (page_graph != nullptr) {
+      std::vector<const String> args;
+      args.push_back(text);
+      page_graph->RegisterWebAPICall("CanvasRenderingContext2D.measureText",
+        args);
+
+      std::stringstream result_buffer;
+      result_buffer << "width: " << result->width()
+        << ", actualBoundingBoxLeft: " << result->actualBoundingBoxLeft()
+        << ", actualBoundingBoxRight: " << result->actualBoundingBoxRight()
+        << ", fontBoundingBoxAscent: " << result->fontBoundingBoxAscent()
+        << ", fontBoundingBoxDescent: " << result->fontBoundingBoxDescent()
+        << ", actualBoundingBoxAscent: " << result->actualBoundingBoxAscent()
+        << ", actualBoundingBoxDescent: " << result->actualBoundingBoxDescent()
+        << ", emHeightAscent: " << result->emHeightAscent()
+        << ", emHeightDescent: " << result->emHeightDescent();
+
+      Baselines* baselines = result->getBaselines();
+      result_buffer << ", hangingBaseline: " << baselines->hanging()
+        << ", alphabeticBaseline: " << baselines->alphabetic()
+        << ", ideographicBaseline: " << baselines->ideographic();
+
+      String page_graph_result(result_buffer.str().c_str());
+      page_graph->RegisterWebAPIResult("CanvasRenderingContext2D.measureText",
+        page_graph_result);
+    }
+
+    return result;
+  }
+#endif
+
   TextMetrics* text_metrics = MakeGarbageCollected<TextMetrics>(
       font, direction, GetState().GetTextBaseline(), GetState().GetTextAlign(),
       text);
