diff --git a/ui/native_theme/native_theme_mac.mm b/ui/native_theme/native_theme_mac.mm
index dab062cf70a36b1631e2aa40131ec30d97dd2099..c5f6fe3f2da423137933dc4706e31c2540016e0b 100644
--- a/ui/native_theme/native_theme_mac.mm
+++ b/ui/native_theme/native_theme_mac.mm
@@ -175,6 +175,7 @@ base::Optional<SkColor> NativeThemeMac::GetOSColor(
 
   // Even with --secondary-ui-md, menus use the platform colors and styling, and
   // Mac has a couple of specific color overrides, documented below.
+  GET_BRAVE_COLOR(color_id)
   switch (color_id) {
     case kColorId_EnabledMenuItemForegroundColor:
       return skia::NSSystemColorToSkColor([NSColor controlTextColor]);
